config/types.go

```diff
@@ -0,0 +1,126 @@
+			Partitions []struct {
+				Label    string `yaml:"label"`
+				Number   int    `yaml:"number"`
+				Size     string `yaml:"size"`
+				Start    string `yaml:"start"`
+				TypeGUID string `yaml:"type_guid"`
+			} `yaml:"partitions"`
+		} `yaml:"disks"`
+		Arrays []struct {
+			Name    string   `yaml:"name"`
+			Level   string   `yaml:"level"`
+			Devices []string `yaml:"devices"`
+			Spares  int      `yaml:"spares"`
+		} `yaml:"raid"`
+		Filesystems []struct {
+			Name  string `yaml:"name"`
+			Mount *struct {
+				Device string `yaml:"device"`
+				Format string `yaml:"format"`
+				Create *struct {
+					Force   bool     `yaml:"force"`
+					Options []string `yaml:"options"`
+				} `yaml:"create"`
+			} `yaml:"mount"`
+			Path string `yaml:"path"`
+		} `yaml:"filesystems"`
+		Files []struct {
+			Filesystem string `yaml:"filesystem"`
+			Path       string `yaml:"path"`
+			Contents   struct {
+				Remote struct {
+					Url          string       `yaml:"url"`
+					Compression  string       `yaml:"compression"`
+					Verification Verification `yaml:"verification"`
+				} `yaml:"remote"`
+				Inline string `yaml:"inline"`
+			} `yaml:"contents"`
+			Mode int `yaml:"mode"`
+			User struct {
+				Id int `yaml:"id"`
+			} `yaml:"uid"`
+			Group struct {
+				Id int `yaml:"id"`
+			} `yaml:"gid"`
+		} `yaml:"files"`
+	} `yaml:"storage"`
+	Systemd struct {
+		Units []struct {
+			Name     string `yaml:"name"`
+			Enable   bool   `yaml:"enable"`
+			Mask     bool   `yaml:"mask"`
+			Contents string `yaml:"contents"`
+			DropIns  []struct {
+				Name     string `yaml:"name"`
+				Contents string `yaml:"contents"`
+			} `yaml:"dropins"`
+		} `yaml:"units"`
+	} `yaml:"systemd"`
+	Networkd struct {
+		Units []struct {
+			Name     string `yaml:"name"`
+			Contents string `yaml:"contents"`
+		} `yaml:"units"`
+	} `yaml:"networkd"`
+	Passwd struct {
+		Users []struct {
+			Name              string   `yaml:"name"`
+			PasswordHash      string   `yaml:"password_hash"`
+			SSHAuthorizedKeys []string `yaml:"ssh_authorized_keys"`
+			Create            *struct {
+				Uid          *uint    `yaml:"uid"`
+				GECOS        string   `yaml:"gecos"`
+				Homedir      string   `yaml:"home_dir"`
+				NoCreateHome bool     `yaml:"no_create_home"`
+				PrimaryGroup string   `yaml:"primary_group"`
+				Groups       []string `yaml:"groups"`
+				NoUserGroup  bool     `yaml:"no_user_group"`
+				System       bool     `yaml:"system"`
+				NoLogInit    bool     `yaml:"no_log_init"`
+				Shell        string   `yaml:"shell"`
+			} `yaml:"create"`
+		} `yaml:"users"`
+		Groups []struct {
+			Name         string `yaml:"name"`
+			Gid          *uint  `yaml:"gid"`
+			PasswordHash string `yaml:"password_hash"`
+			System       bool   `yaml:"system"`
+		} `yaml:"groups"`
+	} `yaml:"passwd"`
+}
+
+type ConfigReference struct {
+	Source       string       `yaml:"source"`
+	Verification Verification `yaml:"verification"`
+}
+
+type Verification struct {
+	Hash *string `yaml:"hash"`
```

Comment:
This would default to sha512 instead of requiring the Hash `Function` and `Sum` be specified each time?